core/src/main/scala/org/apache/spark/deploy/security/HBaseDelegationTokenProvider.scala

```diff
@@ -97,7 +97,8 @@ private[security] class HBaseDelegationTokenProvider
       creds.addToken(token.getService, token)
     } catch {
       case NonFatal(e) =>
-        logWarning(s"Failed to get token from service $serviceName", e)
+        logWarning(s"Failed to get token from service $serviceName due to $e. " +
+          s"If HBase is not used, set spark.security.credentials.hbase.enabled to false")
```

Comment:
When this is explicitly enabled, users or admins would likely know how to disable this .. I am not very sure if this is worthwhile. Stacktrace might be a bit noisy but ideally HBase other other services should not be down if they are in use, and should be up as soon as possible. Given that this warning is not thrown by default easily, I think being noisy here is a-okay. I don't feel strongly on this.

Code-wise, we'll have to fix other instances too together:

```
core/src/main/scala/org/apache/spark/deploy/security/HBaseDelegationTokenProvider.scala:        logWarning(s"Failed to get token from service $serviceName due to  " + e +
core/src/main/scala/org/apache/spark/deploy/security/HBaseDelegationTokenProvider.scala:        logWarning(s"Failed to get token from service $serviceName", e)
core/src/main/scala/org/apache/spark/deploy/security/HadoopFSDelegationTokenProvider.scala:        logWarning(s"Failed to get token from service $serviceName", e)
external/kafka-0-10-token-provider/src/main/scala/org/apache/spark/kafka010/KafkaDelegationTokenProvider.scala:            logWarning(s"Failed to get token from service: $serviceName " +
sql/hive/src/main/scala/org/apache/spark/sql/hive/security/HiveDelegationTokenProvider.scala:        logWarning(s"Failed to get token from service $serviceName", e)
```

and here we might have to use the configuration template (e.g., at `HadoopDelegationTokenManager`) and `serviceName` instead of hard-coding `spark.security.credentials.hbase.enabled`.

Lastly, we would have tho show the error message (`e.getMessage`) and exception type together.

Reply:
In a multi-tenant cluster with HBase service, which is down for some planned maintenance,  even if some teams are not using HBase in their business application, still they are seeing an exception stack trace of more than 100+ line. So it is good to suppress full stack trace and multiple HBase connection failure errors.

> Code-wise, we'll have to fix other instances too together:

1) 

> core/src/main/scala/org/apache/spark/deploy/security/HBaseDelegationTokenProvider.scala

Done

2)
> core/src/main/scala/org/apache/spark/deploy/security/HadoopFSDelegationTokenProvider.scala:        logWarning(s"Failed to get token from service $serviceName", e)

In HadoopFSDelegationTokenProvider.scala, if DT is not generated, application submit will fail fast, hence adding the suggestion to disable HDFS DT generation won't be helpful
3)
> external/kafka-0-10-token-provider/src/main/scala/org/apache/spark/kafka010/KafkaDelegationTokenProvider.scala:            logWarning(s"Failed to get token from service: $serviceName " +

Done
4) 
> sql/hive/src/main/scala/org/apache/spark/sql/hive/security/HiveDelegationTokenProvider.scala:        logWarning(s"Failed to get token from service $serviceName", e)
Done